src/components/time-picker/time-picker.tsx

```diff
@@ -0,0 +1,374 @@
+import * as keycode from 'keycode';
+import * as React from 'react';
+import {SBComponent} from 'stylable-react-component';
+import {Stepper} from '../stepper';
+import styles from './time-picker.st.css';
+import {
+    Ampm, ampmLabels, Format,
+    formatTimeChunk, getCircularValue, is12TimeFormat, isTimeSegment,
+    isTouch, isValidValue, Segment, TimeSegment, to24, toAmpm
+} from './utils';
+
+export interface Props {
+    value?: string;
+    onChange?: (value: string) => void;
+    format?: Format;
+    placeholder?: string;
+    disabled?: boolean;
+    label?: string;
+    name?: string;
+}
+
+export interface State {
+    focus: boolean;
+    hh?: string;
+    mm?: string;
+    format: Format;
+    ampm: Ampm;
+    currentSegment: Segment;
+}
+
+const ampmSwitch = {
+    [Ampm.AM]: Ampm.PM,
+    [Ampm.PM]: Ampm.AM,
+    [Ampm.NONE]: Ampm.NONE
+};
+const segments: Segment[] = ['hh', 'mm', 'ampm'];
+
+function propsValueToSegments(value?: string, format?: Format): {hh?: string, mm?: string, ampm: Ampm} {
+    const isAmpm = format === 'ampm';
+    if (!value) {
+        return {
+            ampm: isAmpm ? Ampm.AM : Ampm.NONE
+        };
+    }
+    const [hh24, mm] = value.split(':').map(Number);
+    const {hh, ampm} = toAmpm(hh24);
+    return {
+        mm: formatTimeChunk(mm),
+        hh: formatTimeChunk(isAmpm ? hh : hh24),
+        ampm: isAmpm ? ampm : Ampm.NONE
+    };
+}
+
+@SBComponent(styles)
+export default class TimePicker extends React.Component<Props, State> {
+    public static defaultProps = {
+        format: is12TimeFormat ? 'ampm' : '24hr',
+        disabled: false
+    };
+    private segments: {
+        [key: string]: HTMLInputElement | HTMLDivElement | null
```

Comment:
yes, ideally it should be
```ts
[key: Segment]: HTMLInputElement | HTMLDivElement | null
```
but typescript does not allow to use any type except `string` and `number` here.
I will try to find better solution if any